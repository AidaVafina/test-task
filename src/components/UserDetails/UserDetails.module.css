.container {
    background-color: var(--color-card-bg, #fff);
    border: 1px solid var(--color-border, #e0e0e0);
    border-radius: 8px;
    padding: 1.5rem;
    box-shadow: var(--shadow, 0 2px 4px rgba(0, 0, 0, 0.1));
    margin-bottom: 1rem;
}

.name {
    margin: 0 0 1rem 0;
    font-size: 1.5rem;
    color: var(--color-text-primary, #333);
}

.item {
    margin: 0.5rem 0;
    font-size: 1rem;
    color: var(--color-text-secondary, #555);
}

.link {
    color: var(--color-primary, #3498db);
    text-decoration: none;
}

.link:hover {
    text-decoration: underline;
}
