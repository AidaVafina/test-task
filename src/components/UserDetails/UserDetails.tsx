import React from 'react';
import styles from './UserDetails.module.css';
import type {User} from "../../types";

interface UserDetailsProps {
    user: User;
}

const UserDetails: React.FC<UserDetailsProps> = ({ user }) => {
    return (
        <div className={styles.container}>
            <h2 className={styles.name}>{user.name}</h2>
            <p className={styles.item}>
                <strong>Email: </strong>
                <a href={`mailto:${user.email}`} className={styles.link}>
                    {user.email}
                </a>
            </p>
            <p className={styles.item}>
                <strong>Телефон: </strong>
                <a href={`tel:${user.phone}`} className={styles.link}>
                    {user.phone}
                </a>
            </p>
            <p className={styles.item}>
                <strong>Адрес: </strong>
                {`${user.address.suite}, ${user.address.street}, ${user.address.city}, ${user.address.zipcode}`}
            </p>
            <p className={styles.item}>
                <strong>Компания: </strong>
                {user.company.name}
            </p>
        </div>
    );
};

export default UserDetails;
