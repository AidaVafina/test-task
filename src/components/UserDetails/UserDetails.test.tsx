import { render } from '@testing-library/react';
import UserDetails from './UserDetails';

const mockUser = {
    id: 1,
    name: 'John Doe',
    username: 'johndoe',
    email: 'john@example.com',
    address: {
        city: 'New York',
        street: '123 Main St',
        suite: 'Apt 4B',
        zipcode: '10001',
        geo: {
            lat: '40.7128',
            lng: '-74.0060'
        }
    },
    phone: '123-456-7890',
    website: 'johndoe.com',
    company: {
        name: 'Test Company',
        catchPhrase: 'Test catch phrase',
        bs: 'Test bs'
    }
};

describe('UserDetails Component', () => {
    it('should match snapshot', () => {
        const { container } = render(<UserDetails user={mockUser} />);
        expect(container).toMatchSnapshot();
    });

    it('should render user information correctly', () => {
        const { container } = render(<UserDetails user={mockUser} />);

        expect(container.textContent).toContain(mockUser.name);
        expect(container.textContent).toContain(mockUser.email);
        expect(container.textContent).toContain(mockUser.phone);
        expect(container.textContent).toContain(mockUser.address.city);
        expect(container.textContent).toContain(mockUser.company.name);
    });
}); 