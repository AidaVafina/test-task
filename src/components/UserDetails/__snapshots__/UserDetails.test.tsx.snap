// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserDetails Component > should match snapshot 1`] = `
<div>
  <div
    class="_container_ed71f6"
  >
    <h2
      class="_name_ed71f6"
    >
      John Doe
    </h2>
    <p
      class="_item_ed71f6"
    >
      <strong>
        Email: 
      </strong>
      <a
        class="_link_ed71f6"
        href="mailto:john@example.com"
      >
        john@example.com
      </a>
    </p>
    <p
      class="_item_ed71f6"
    >
      <strong>
        Телефон: 
      </strong>
      <a
        class="_link_ed71f6"
        href="tel:123-456-7890"
      >
        123-456-7890
      </a>
    </p>
    <p
      class="_item_ed71f6"
    >
      <strong>
        Адрес: 
      </strong>
      Apt 4B, 123 Main St, New York, 10001
    </p>
    <p
      class="_item_ed71f6"
    >
      <strong>
        Компания: 
      </strong>
      Test Company
    </p>
  </div>
</div>
`;

exports[`UserDetails Component should match snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <h2
      class="name"
    >
      John Doe
    </h2>
    <p
      class="item"
    >
      <strong>
        Email: 
      </strong>
      <a
        class="link"
        href="mailto:john@example.com"
      >
        john@example.com
      </a>
    </p>
    <p
      class="item"
    >
      <strong>
        Телефон: 
      </strong>
      <a
        class="link"
        href="tel:123-456-7890"
      >
        123-456-7890
      </a>
    </p>
    <p
      class="item"
    >
      <strong>
        Адрес: 
      </strong>
      Apt 4B, 123 Main St, New York, 10001
    </p>
    <p
      class="item"
    >
      <strong>
        Компания: 
      </strong>
      Test Company
    </p>
  </div>
</div>
`;
